src/discordcr/rest.cr

```diff
@@ -132,15 +144,16 @@ module Discord
     # permission.
     #
     # [API docs for this method](https://discordapp.com/developers/docs/resources/channel#modify-channel)
-    def modify_channel(channel_id : UInt64, name : String?, position : UInt32?,
-                       topic : String?, bitrate : UInt32?, user_limit : UInt32?)
-      json = {
-        name:       name,
-        position:   position,
-        topic:      topic,
-        bitrate:    bitrate,
-        user_limit: user_limit,
-      }.to_json
+    def modify_channel(channel_id : UInt64, name : String? = nil, position : UInt32? = nil,
+                       topic : String? = nil, bitrate : UInt32? = nil, user_limit : UInt32? = nil,
+                       nsfw : Bool? = nil)
+      json = encode_tuple(
```

Comment:
This can't compile, because you're not passing a namedtuple into `encode_tuple`.

Reply:
I had fixed but hadn't pushed it ;(

Is the double-splat acceptable or would you actually prefer the restrictions & pass named tuple literals? I'm not fully aware of how the compiler types `**args` / what difference it would make.